pyproject.toml

```diff
@@ -84,6 +85,9 @@ cache_dir = ".cache/pytest"
 branch = true
 source = ["app"]
 relative_files = true
+# gevent is needed for fastapi projects with async/await calls, otherwise the coverage misses them
+# ref: https://stackoverflow.com/a/76033894
+concurrency = ["gevent"]
```

Comment:
You can use greenlet here and avoid installing gevent.
Looks like SQLAlchemy uses greenlets to add asyncio support.

Reply:
I have a question: 
I'd like to understand why we need Gevent here. 
Gevent is a library for green threads. It is typically used in projects relying on synchronous code where concurrency is desired. FastAPI already embraces asyncio, making Gevent redundant, in my opinion.
How could it directly impact test coverage?

Reply:
In fact is not related to fastapi. is related to sqlalchemy. sqlalchemy already depends on greenlet library so the only thing that is needed to collect the coverage the right way is to specify to coverage the concurrency libraries used in the project (see [https://coverage.readthedocs.io/en/7.6.9/config.html#run-concurrency](https://coverage.readthedocs.io/en/7.6.9/config.html#run-concurrency). Probably the right solution here is to specify both greenlet and thread:

```
concurrency = ["thread", "greenlet"]
```

Reply:
Thank you for this guys, learning new things every day :) Changed to `["thread", "greenlet"]` and updated the comment to be more accurate, also included these links